azurecompute-arm/src/main/java/org/jclouds/azurecompute.arm/features/JobApi.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute.arm.features;
+import java.io.Closeable;
+import java.net.URI;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.Path;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.GET;
+import org.jclouds.oauth.v2.filters.OAuthFilter;
+import org.jclouds.rest.annotations.RequestFilters;
+import org.jclouds.rest.annotations.ResponseParser;
+import org.jclouds.azurecompute.arm.functions.ParseJobStatus;
+import org.jclouds.azurecompute.arm.functions.ParseJobStatus.JobStatus;
+
+/**
+ * The Azure Resource Manager API checks for job status and progress.
+ */
+@Path("/")
+
+@RequestFilters(OAuthFilter.class)
+@Consumes(MediaType.APPLICATION_JSON)
+public interface JobApi extends Closeable{
+   @GET
+   @ResponseParser(ParseJobStatus.class)
+   JobStatus jobDone(URI jobURI, @QueryParam("api-version") String apiversion);
```

Comment:
I think it would be better to annotate the jobURI parameter with `@EndpointParam`, and remove the `@Path` annotation from the class, to designate that the endpoint will be computed entirely based on the uri parameter.
Also rename the method to `jobStatus`?